docs/metrics.md

```diff
@@ -116,6 +224,58 @@ metrics = context.metrics
 
 All metrics are collected within the context.
 
+
+### Specifying Metric Types
+
+When adding any metric via Metrics API, users have the ability to override the metric type by specifying the positional argument
+`metric_type=MetricTypes.[counter/gauge/histogram]`.
+
+```python
+metrics.add_metric("GenericMetric", value=1, ..., metric_type=MetricTypes.gauge)
```

Comment:
Is `add_metric()` the right name? This feels more like an `add_measurement()` to me otherwise why we do we need to setup a yaml file?